src/components/Minecraft/MinecraftCreateForm.tsx

```diff
@@ -1,72 +1,122 @@
-import { MinecraftFlavour } from 'bindings/MinecraftFlavour';
-import { MinecraftSetupConfigPrimitive } from 'bindings/MinecraftSetupConfigPrimitive';
 import Button from 'components/Atoms/Button';
 import { Form, Formik, FormikHelpers, FormikProps } from 'formik';
-import { useRef, useState } from 'react';
+import { useRef, useState, useEffect } from 'react';
 import { useEffectOnce } from 'usehooks-ts';
 import useAnalyticsEventTracker from 'utils/hooks';
 import { axiosWrapper } from 'utils/util';
 import {
+  ConfigurableValue,
   formId,
-  initialValues,
-  MinecraftSetupConfigPrimitiveForm,
-  validationSchema,
+  ManifestValue,
+  ConfigurableManifest,
+  SectionManifestValue,
 } from './Create/form';
-import MinecraftAdvancedForm from './Create/MinecraftAdvancedForm';
-import MinecraftBasicForm from './Create/MinecraftBasicForm';
-import MinecraftNameForm from './Create/MinecraftNameForm';
-import { flavourStringToMinecraftFlavour } from 'bindings/impl/FlavourStringToMinecraftFlavour';
-
-function _renderStepContent(step: number) {
-  switch (step) {
-    case 0:
-      return <MinecraftNameForm />;
-    case 1:
-      return <MinecraftBasicForm />;
-    case 2:
-      return <MinecraftAdvancedForm />;
-    default:
-      return 'Unknown step';
+import { generateValidationSchema, generateInitialValues } from './Create/form';
+import { createForm } from './Create/FormCreation';
+import GameTypeSelectForm from './Create/GameTypeSelectForm';
+import { SetupInstanceManifest } from 'data/InstanceGameTypes';
+import { HandlerGameType } from 'bindings/HandlerGameType';
+import Spinner from 'components/DashboardLayout/Spinner';
+import clsx from 'clsx';
+
+function _renderStepContent(
+  step: number,
+  gameType: HandlerGameType,
+  setGameType: (gameType: HandlerGameType) => void,
+  setupManifest?: ConfigurableManifest | null
+) {
+  if (!setupManifest) {
+    return null;
   }
-}
+  const forms = Object.keys(setupManifest['setting_sections']).map(
```

Comment:
Might want to use `useMemo` on this, or generate the form using another way.
Currently this get's called every render. Thus every time the user presses a key on their keyboard we recreate the forms again